doc/api.yaml

```diff
@@ -0,0 +1,113 @@
+openapi: 3.0.2
+
+info:
+  title: MockIO
+  version: 1.0.0
+  contact:
+    name: Source Busters
+  description: |
+    Configurable mock server for mocking HTTP requests.
+
+
+paths:
+  /config:
+
+    post:
+      summary: Create a named configurable mock instance
+
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/MockParams"
+
+      responses:
+        "204":
+          description: Mock instance was created successfully
+
+        "404":
+          description: Mock parameters validation failed
+
+
+
+  /config/{mockId}:
+
+      delete:
+        summary: Resets a named configurable mock instance
+      responses:
+        "204":
+          description: the mock was reseted successfully
+
+
+      post:
+        summary: Creates a request matcher configuration
+
+        requestBody:
+          required: true
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/MatcherParams"
+
+  /{mockType}/{mockId}:
+    summary: Mock instance root path. A dynamic endpoint which requires a mock configuration to be created first
+
+
+components:
+  parameters:
+    mockId:
+      description: >
+        The unique mock configuration identifier.
+      name: mockId
+      in: path
+      required: true
+      schema:
+        $ref: "#/components/schemas/mockId"
+
+    mockType:
+      description: >
+        The mock configuration type.
+      name: mockId
+      in: path
+      required: true
+      schema:
+        $ref: "#/components/schemas/mockType"
+
+  schemas:
+    mockId:
```

Comment:
instead of being an opaque identifier, wouldn't it be better to use a meaningful name as the identifier,  such as the service you're mocking?

Reply:
that should be user's decision, i just specified it's an alphanumeric string